tests/defaultexecutor.cpp

```diff
@@ -180,6 +217,45 @@ TEST_F(DefaultExecutorTest, ThenWithException)
     executor->stop();
 }
 
+using DefaultExecutorDeathTest = DefaultExecutorTest;
+
+TEST_F(DefaultExecutorDeathTest, ObserveWithException)
+{
+    GTEST_FLAG_SET(death_test_style, "threadsafe");
+
+    auto executor = make_shared<DefaultExecutor>(milliseconds(10));
+    Default<Executor>::Setter execSetter(executor);
+
+    mutex mtx;
+    condition_variable cv;
+
+    // This will never become true, as the continuation will rethrow the exception
```

Comment:
Can we add a `EXPECT_FALSE(success);` to the end of the test to make this expectation checked?